make/autoconf/flags-cflags.m4

```diff
@@ -134,6 +134,7 @@ AC_DEFUN([FLAGS_SETUP_WARNINGS],
 
       WARNINGS_ENABLE_ALL="-W3"
       DISABLED_WARNINGS="4800"
+      DISABLED_WARNINGS+=" 4146" # unary minus operator applied to unsigned type, result still unsigned
```

Comment:
This change will affect *all* JDK code. I'm not sure this was intended?

If it was intended, I think you need to motivate this more explicitly.

If you only wanted to disable the warning for hotspot, the proper solution would be to add it to DISABLED_WARNINGS_microsoft in make/hotspot/lib/CompileJvm.gmk.

Reply:
Thank you @magicus! It was indeed meant only for the hotspot part.